api/compaction_manager.cc

```diff
@@ -119,7 +119,7 @@ void set_compaction_manager(http_context& ctx, routes& r) {
             auto& cm = db.get_compaction_manager();
             return parallel_for_each(table_names, [&db, &cm, &ks_name, type] (sstring& table_name) {
                 auto& t = db.find_column_family(ks_name, table_name);
-                return cm.stop_compaction(type, &t);
+                return cm.stop_compaction(type, &t.as_table_state());
```

Comment:
I think that the compaction_manager can still expose `stop_compaction(table&)`
and we can have a variant of stop_ongoing_compactions that will still get a table*
and compare it the tasks' via each task compacting_table().

The per- table_state / compaction_group variant can stop ongoing tasks using
a respective task::compaction_group().

The reason is we don't have to expose the complexity to the caller, and force the callers (api, table) to iterate through the table compaction groups to stop each one of them - requiring a full traversal of the compaction_manager tasks for each.  We better traverse it once and look up the table.

Reply:
Yes, initially I thought of adding a stop_compaction() variant that accepts ks_name and cf_name, so APIs can use it to stop tasks running on a replica::table. But then I realized that mixing replica::table and compaction::table_state would bring lots of complexity to compaction_manager. By limiting interaction with manager to table_state, we have a more solid interface.